src/us/kbase/workspace/test/database/UtilTest.java

```diff
@@ -0,0 +1,222 @@
+        for (String nullOrEmpty : nonEmptyStrings) {
+            assertThat(INCORRECT_NULL_EMPTY, Util.isNullOrEmpty(nullOrEmpty), is(false));
+        }
+    }
+
+    @Test
+    public void passNonNull() throws Exception {
+        for (String empty : emptyStrings) {
+            Util.nonNull(empty, OK_STRING);
+        }
+
+        for (String nonEmpty : nonEmptyStrings) {
+            Util.nonNull(nonEmpty, OK_STRING);
+        }
+    }
+
+    @Test
+    public void failNonNull() throws Exception {
+        try {
+            Util.nonNull(null, OK_STRING);
+            fail(EXP_EXC);
+        } catch (Exception got) {
+            TestCommon.assertExceptionCorrect(got, new NullPointerException(OK_STRING));
+        }
+    }
+
+    @Test
+    public void passNoNulls() throws Exception {
+        Util.noNulls(nonEmptyStrings, TYPE_NAME);
+        Util.noNulls(emptyStrings, TYPE_NAME);
+    }
+
+    @Test
+    public void failNoNulls() throws Exception {
+        final List<List<String>> testList = new ArrayList<>();
+        testList.add(emptyStringsWithNull);
+        testList.add(nonEmptyStringsWithNull);
+
+        for (List<String> testListItem : testList) {
+            try {
+                Util.noNulls(testListItem, OK_STRING);
+                fail(EXP_EXC);
+            } catch (Exception got) {
+                TestCommon.assertExceptionCorrect(got, new NullPointerException(OK_STRING));
+            }
+        }
+    }
+
+    @Test
+    public void passCheckNoNullsOrEmpties() throws Exception {
+        Util.checkNoNullsOrEmpties(nonEmptyStrings, TYPE_NAME);
+    }
+
+    @Test
+    public void failCheckNoNullsOrEmpties() throws Exception {
+        final List<List<String>> testList = new ArrayList<>();
+        testList.add(emptyStrings);
+        testList.add(emptyStringsWithNull);
+        testList.add(nonEmptyStringsWithNull);
+
+        for (List<String> testListItem : testList) {
+            try {
+                Util.checkNoNullsOrEmpties(testListItem, TYPE_NAME);
+                fail(EXP_EXC);
+            } catch (Exception got) {
+                TestCommon.assertExceptionCorrect(got, new IllegalArgumentException(
+                        "Null or whitespace only string in collection " + TYPE_NAME));
+            }
+        }
+    }
+
+    @Test
+    public void passCheckString() throws Exception {
+        final String INCORRECT_CHECKSTRING = "incorrect checkString";
+        final Map<String, String> trimmedNonEmptyStrings = new HashMap<>();
+        for (String nonEmpty : nonEmptyStrings) {
+            trimmedNonEmptyStrings.put(nonEmpty, nonEmpty.trim());
+        }
+
+        for (Map.Entry<String, String> mapElement : trimmedNonEmptyStrings.entrySet()) {
+            assertThat(
+                    INCORRECT_CHECKSTRING,
+                    Util.checkString(mapElement.getKey(), TYPE_NAME),
+                    is(mapElement.getValue()));
+        }
+
+        final int max = 100;
```

Comment:
I would make this one larger than the largest string in your test set

Reply:
Fixed

Reply:
:+1: